contracts/cobalt/LoanEngine.sol

```diff
@@ -422,62 +432,88 @@ contract LoanEngine is Ownable, ERC721Base {
     function pay(uint256 loanId, uint128 amount, address from, bytes oracleData) external returns (bool) {
         Loan storage loan = loans[loanId];
         require(loan.status == Status.ongoing, "The loan is not ongoing");
-        moveCheckpoint(loan, uint64(now));
-        if (loan.status == Status.ongoing) {
-            uint128 available = amount;
-            uint128 unpaidInterest;
-            uint128 pending;
-            uint128 target;
-            do {
-                // Pay the full installment or the max ammount possible
-                pending = uint128(_currentDebt(loan));
-                target = pending < available ? pending : available;
-
-                // Calc paid base
-                unpaidInterest = loan.interest - (loan.paid - loan.paidBase);
-                loan.paidBase += target > unpaidInterest ? target - unpaidInterest : 0;
-
-                loan.paid += target;
-                loan.lenderBalance += target;
-                available -= target;
-
-                emit PartialPayment(loanId, msg.sender, from, target, unpaidInterest);
-
-                // If the loan is fully paid stop paying
-                if (checkFullyPaid(loan)) {
-                    break;
-                }
-
-                // If current installment was fully paid move to the next one
-                if (pending == target) {
-                    advanceClock(loan, loan.installmentDuration);
-                }
-            } while (available != 0);
-
-            uint256 requiredTransfer = convertRate(loan.oracle, loan.currency, oracleData, amount - available);
-            require(token.transferFrom(msg.sender, this, requiredTransfer), "Error pulling tokens");
+
+        uint128 available = payProgresive(loan, amount, loanId, from);
+
+        if (loan.oracle == address(0)) {
+            available = amount - available;
+        } else {
+            uint rate;
+            uint decimals;
```

Comment:
The rest of the contract uses uint256 and not the alias uint, also it can could be replaced by
(uint256 rate, uint256 decimals) = ...

Reply:
I have an old version of solidity. 
Now I update and use uint256